src/main/java/com/adobe/target/edge/client/service/DefaultTargetService.java

```diff
@@ -45,6 +48,9 @@
   private final TargetHttpClient targetHttpClient;
   private final ClientConfig clientConfig;
   private String stickyLocationHint;
+  private final TelemetryService telemetryService;
+
+  private final ConcurrentLinkedQueue<TelemetryEntry> storedTelemetries = new ConcurrentLinkedQueue<>();
```

Comment:
Let's encapsulate all telemetry-related internals in `TelemetryService` or `TelemetryProvider` as specified in the Phase II Telemetry Wiki